jsonschema-core.xml

```diff
@@ -324,10 +309,13 @@
 
         <section title="Schema references with $ref">
             <t>
-                In addition to a boolean value or an object using schema kewyords defined
-                in the meta-schema, a schema may be represnted by an object containing a "$ref" property.
-                The value of the $ref is a URI Reference.
+                The "$ref" keyword is used to reference an externally defined schema, and provides the ability to validate recursive structures through self-reference.
+            </t>
+            <t>
+                An object subschema with a "$ref" property MUST be interpreted as a "$ref" reference.
+                The value of the "$ref" property MUST be a URI Reference.
                 Resolved against the current URI base, it identifies the URI of a schema to use.
+                The referenced schema MUST NOT itself contain a "$ref" property.
```

Comment:
I'm not sure I follow here- when we spoke on IRC the idea was that both booleans and references would be valid schemas anywhere.  It's a bit weird to have a schema that's just a reference but there could be a reason for it.  I still think SHOULD NOT is more correct, but I think I might be missing part of your point.